datanode/entities/aggregated_ledgerentries.go

```diff
@@ -0,0 +1,103 @@
+package entities
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/types"
+	v2 "code.vegaprotocol.io/vega/protos/data-node/api/v2"
+	"code.vegaprotocol.io/vega/protos/vega"
+	"github.com/shopspring/decimal"
+)
+
+// AggregatedLedgerEntries represents the the summed amount of ledger entries for a set of accounts withing a given time range.
+// VegaTime and Quantity will always be set. The others will be nil unless when
+// querying grouping by one of the corresponding fields is requested.
+type AggregatedLedgerEntries struct {
+	ID       *int64
+	VegaTime time.Time
+
+	AccountFromID AccountID
+	AccountToID   AccountID
+
+	Quantity decimal.Decimal
+
+	TransferType *LedgerMovementType
+	AccountType  *types.AccountType
+
+	PartyID  PartyID
+	AssetID  AssetID
+	MarketID MarketID
+}
+
+func (ledgerEntries *AggregatedLedgerEntries) ToProto() *v2.AggregatedLedgerEntries {
+	lep := &v2.AggregatedLedgerEntries{}
+
+	accountFromIDString := ledgerEntries.AccountFromID.String()
+	if accountFromIDString != "" {
+		lep.AccountFromId = &accountFromIDString
+	}
+
+	accountToIDString := ledgerEntries.AccountToID.String()
+	if accountToIDString != "" {
+		lep.AccountToId = &accountFromIDString
+	}
+
+	partyIDString := ledgerEntries.PartyID.String()
+	if partyIDString != "" {
+		lep.PartyId = &partyIDString
+	}
+
+	assetIDString := ledgerEntries.AssetID.String()
+	if assetIDString != "" {
+		lep.AssetId = &assetIDString
+	}
+
+	marketIDString := ledgerEntries.MarketID.String()
+	if marketIDString != "" {
+		lep.MarketId = &marketIDString
+	}
+
+	if ledgerEntries.AccountType != nil {
+		lep.AccountType = *ledgerEntries.AccountType
+	}
+
+	if ledgerEntries.TransferType != nil {
+		lep.TransferType = vega.TransferType(*ledgerEntries.TransferType)
+	}
+
+	return lep
+}
+
+func (ledgerEntries AggregatedLedgerEntries) Cursor() *Cursor {
+	return NewCursor(AggregatedLedgerEntriesCursor{
+		VegaTime: ledgerEntries.VegaTime,
```

Comment:
VegaTime is the block time, so it is sufficient to display a single or a group of entries withing the same block.
If we have 20 entries in a block and do not group them - we will have one row with quantity - sum of all their quantities.
If we start grouping them - yes it can become pretty complex.
I did it in the way `AggregatedBalances` are done; if we remove the grouping - we will simply have a list by vegatime, but do we want to _know_ which of the entries are of some transfer type for example?

Reply:
I change AggregatedBalances recently for that reason, if you look now you'll see that the sort order and what is in the cursor depends on what you're grouping by.. the problem comes if you're paging and the page boundary doesn't align with a change in VegaTime - e.g. if you were grouping by party you might have

```
vega_time party_id amount
1         a        1
1         b        2
1         c        3
2         a        10
```

So you can see that vega_time is not sufficient to identify which row you're on - if the page size was 2 the current cursor cannot reflect that you got up to `{vega_time, party: b}

Reply:
So you suggest to remove it, is that correct?

Reply:
okay